dtfs-central-api/src/v1/controllers/utilisation-report-service/get-selected-fee-records-details.controller/helpers.test.ts

```diff
@@ -182,11 +192,112 @@ describe('get selected fee record details controller helpers', () => {
       const canAddToExistingPayment = true;
 
       // Act
-      const result = await mapToSelectedFeeRecordDetails(bankId, aReportPeriod(), [aFeeRecordWithStatusToDo()], canAddToExistingPayment);
+      const result = await mapToSelectedFeeRecordDetailsWithoutAvailablePaymentGroups(
+        bankId,
+        aReportPeriod(),
+        [aFeeRecordWithStatusToDo()],
+        canAddToExistingPayment,
+      );
 
       // Assert
       expect(result.canAddToExistingPayment).toEqual(canAddToExistingPayment);
     });
+
+    describe('when availablePaymentGroups is undefined', () => {
+      it('returns an object which does not contain the available fee record payment groups', async () => {
+        // Arrange
+        const bankId = '123';
+        const canAddToExistingPayment = false;
+
+        // Act
+        const result = await mapToSelectedFeeRecordDetailsWithoutAvailablePaymentGroups(
+          bankId,
+          aReportPeriod(),
+          [aFeeRecordWithStatusToDo()],
+          canAddToExistingPayment,
+        );
+
+        // Assert
+        expect(result.availablePaymentGroups).toBeUndefined();
+      });
+    });
+  });
+
+  describe('getSelectedFeeRecordsAvailablePaymentGroups', () => {
+    it('should return mapped fee record payment groups', async () => {
+      // Arrange
+      const reportId = '123';
+      const paymentCurrency = 'GBP';
+
+      const firstPaymentEntity = PaymentEntityMockBuilder.forCurrency('GBP')
+        .withDateReceived(new Date('2022-01-01'))
+        .withAmount(55)
+        .withId(1)
+        .withReference('First payment')
+        .build();
+      const secondPaymentEntity = PaymentEntityMockBuilder.forCurrency('GBP')
+        .withDateReceived(new Date('2022-01-01'))
+        .withAmount(65)
+        .withId(2)
+        .withReference('Second payment')
+        .build();
+      const thirdPaymentEntity = PaymentEntityMockBuilder.forCurrency('GBP')
+        .withDateReceived(new Date('2022-01-01'))
+        .withAmount(75)
+        .withId(3)
+        .withReference('Third payment')
+        .build();
+
+      const firstFeeRecordEntity = FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build())
+        .withExporter('Test company')
+        .withFacilityId('00012345')
+        .withId(1)
+        .withPayments([firstPaymentEntity, secondPaymentEntity])
+        .withStatus('DOES_NOT_MATCH')
+        .build();
+      const secondFeeRecordEntity = FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').build())
+        .withExporter('Test company')
+        .withFacilityId('00012345')
+        .withId(2)
+        .withPayments([thirdPaymentEntity])
+        .withStatus('DOES_NOT_MATCH')
+        .build();
```

Comment:
Can we remove the data that is irrelevant to the test? e.g. the payment date is unnecessary and should get set to a default value by the mock entity builder and then the fee records it doesn't really matter what the exporter and facility id etc are I don't think? but it take up a lot of lines of code and distracts from the key information

Reply:
Very good point, fixed this on both the payment and fee record entities thanks